File: javalin5/javalin-testing-example/src/main/java/io/javalin/example/java/UserController.java
package io.javalin.example.java;

import io.javalin.http.BadRequestResponse;
import io.javalin.http.Context;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class UserController {

    static List<String> users = new ArrayList<>(Arrays.asList("User1", "User2", "User3"));

    public static void create(Context ctx) {
        String username = ctx.queryParam("username");
        if (username == null || username.length() < 5) {
            throw new BadRequestResponse();
        } else {
            users.add(username);
            ctx.status(201);
        }
    }

    public static void getAll(Context ctx) {
        ctx.json(users);
    }

}

File: javalin5/javalin-testing-example/src/main/kotlin/io/javalin/example/kotlin/UserController.kt
package io.javalin.example.kotlin

import io.javalin.http.BadRequestResponse
import io.javalin.http.Context

object UserController {

    var users = mutableListOf("User1", "User2", "User3")

    fun create(ctx: Context) {
        val username = ctx.queryParam("username")
        if (username == null || username.length < 5) {
            throw BadRequestResponse()
        } else {
            users.add(username)
            ctx.status(201)
        }
    }

    fun getAll(ctx: Context) {
        ctx.json(users)
    }

}
